### PR TITLE
docs(table): remove react wrapper for table stories

### LINKED ISSUE #3304: [Table] Stories slow rerender
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug
Table stories noticeably rerenders when changing a knob 

See comment
https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3274#issuecomment-1040881902

### To Reproduce

1. Go to https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-table--playground
2. Change any knob 
3. See the slow rerendering

> Please create a reduced test case in CodeSandbox:
> https://codesandbox.io/s/github/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/examples/codesandbox

### Environment/versions:

- OS: [e.g. MacOS, Windows]
- Browser: [e.g. chrome, safari]
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #3304

**Summary**
Our stories using the StatefulTable had become slow due to the fact that the whole story was re-generated each time a knob was changed. This happened when the story was wrapped in a React element using the story `decorator`, which we did for each story that needed DragAndDrop or needed a react state.

**Change List (commits, features, bugs, etc)**
- The Tables are no longer re-generated from scratch when a knob changes. Some knobs still explicitly re-renders the table for demo purposes, e.g. when view related knobs change for the StatefulTable we need to use `key` to get the change picked up.
- Removed DnD wrapper from the decorator and added the `<DragAndDrop>` element in the story itself.
- Created a non-react dependent version of React's `useState` called `useStoryState`. For simplicity of use I built it with the same API as react. In order to support our more advanced stories where different knobs also affects the state I added the option of providing a dependency array that when changed will "reset" the state to the current initial value.

**Acceptance Test (how to verify the PR)**
Not re-generating the story when knob changes has exposed yet more bugs in the table. I have filed issues for the once I have found in my testing but there might be more.

1. Go to https://deploy-preview-3324--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table--playground
2. Click on the Title & Toolbar tab
3. Edit the knob `secondaryTitle` and make sure the table doesn't reload in a noticeable manner
4. Test with more knobs (remember that some knobs explicitly re-renders the table using the key)
5. Test the other "new" table stories to make sure there is no slow re-generation that makes the table flicker when knob changes

**Regression Test (how to make sure this PR doesn't break old functionality)**
https://deploy-preview-3324--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table--playground

All "new" table stories works as expected, no crashes when knobs are changed. Note, not all knobs in the stories are applied when changed when the type of table is StatefulTable (since the state of the knob has moved into the table after the initial render).

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by kevinsperrine — APPROVED** [review]
Awesome work. How did you track this down?

**Comment by bjornalm** [comment]
> Awesome work. How did you track this down?

Thanks, tedious trial and error... we did also have a couple of old StatefulTable stories where this didn't happen so I just had to compare with them to figure out what the difference was. The decorator was not my first guess, I suspected a knob or table setting initially.

File: packages/react/src/internal/storyState.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { FORCE_RE_RENDER, STORY_RENDERED, STORY_CHANGED } from '@storybook/core-events';
+import { isEqual } from 'lodash-es';
+
+let callOrder = 0;
+let states = [];
+let dependencies = [];
+
+addons.getChannel().addListener(STORY_CHANGED, () => {
+  // Clear the global state and reset the capll order when
```
Comment (davidicus) [inline packages/react/src/internal/storyState.js:12]: typo:
```suggestion
  // Clear the global state and reset the call order when
```

File: packages/react/src/internal/storyState.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * and trigger a force rerender of the story to show the modified changes.
+ *
+ * Example 1
+ * The dependencies can be a one or more knob values affecting the initial value like the Table's
+ * 'hasLoadMore' affects the 'data' prop.
+ *
+ * Example 2
+ * The dependencies can be the value of a knob representing the same table prop as the state does
+ * so that that they can be used together, i.e. when the knob is modified the state
+ * is also updatet with the same value.
```
Comment (davidicus) [inline packages/react/src/internal/storyState.js:39]: typo:
```suggestion
 * is also updated with the same value.
```

File: packages/react/src/internal/storyState.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  // Set the value to the initialValue if this is the first render, if not
+  // the value stays unchanged
+  const isInitialRender = states[index] === undefined;
+  const value = isInitialRender ? initialValue : states[index];
+  states[index] = value;
+
+  // If the dependencies has changed the state is "reset" to use the
+  // new initial value and force a rerender of the story just like
+  // a normal update would, thus the resetted value is picked up during
```
Comment (davidicus) [inline packages/react/src/internal/storyState.js:59]: typo: 
```suggestion
  // a normal update would, thus the reset value is picked up during
```

**Review by davidicus — APPROVED** [review]
awesome. great detective work.

**Pull request merged**